### PR TITLE
Remove local.config.php file from usage.

### PR DESCRIPTION
**Pull request by josephsnyder**
Remove references to the config.local.php file from the environment in CDash.  Use the .env file when necessary and trust Laravel to read and refresh as necessary.

### CODE REVIEW

File: app/Console/Commands/MigrateConfig.php [hunk 1]
```diff
@@ -80,7 +80,12 @@ public function handle()
         // Get default values of legacy configuration variables.
         $legacy_defaults = [];
         $ONLY_LOAD_DEFAULTS = true;
-        include $cdash_app_dir . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'config.php';
+        $config_dir = $cdash_app_dir . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR;
+        include $config_dir  . 'config.php';
+        if (file_exists($config_dir . 'config.local.php'))
+        {
+            include $config_dir . 'config.local.php';
+        }
```
Comment (williamjallen) [inline app/Console/Commands/MigrateConfig.php:88]: I believe the current CDash practice is to put the opening curly brace on the same line as the if-statement.  Only function and class declarations have the opening curly brace on a separate line.  The same change applies to the area around line 124 as well.

Our style checker didn't pick this up, so perhaps we should take a look at the rules being enforced to help standardize things like this.

File: tests/Feature/MigrateConfigCommand.php [hunk 2]
```diff
@@ -11,6 +11,10 @@ public function setUp() : void
         parent::setUp();
         $this->config_file = base_path('app/cdash/config/config.local.php');
         $this->test_file = base_path('tests/.env.testing');
+        if (!file_exists($this->config_file))
+        {
+            file_put_contents($this->config_file, "");
+        }
```
[the remark this answers is not in the document]
Comment (josephsnyder) [reply tests/Feature/MigrateConfigCommand.php]: Good catch, thanks!

File: app/Console/Commands/MigrateConfig.php (hunk 1 above)
Comment (josephsnyder) [reply app/Console/Commands/MigrateConfig.php:88]: Thanks!

File: app/cdash/tests/test_enable_async.php [hunk 3]
```diff
@@ -14,24 +14,10 @@ public function __construct()
...
-            }
-            if ($line != '') {
-                fwrite($handle, "$line\n");
-            }
-        }
-        fclose($handle);
-        unset($handle);
+        $filename = dirname(__FILE__) . '/../../../.env';
+        // Using .env, we no longer have to worry about being inside the closing PHP bracket.
+        $injectedText = '// test config settings injected by file [' . __FILE__ . ']\nCDASH_ASYNCHRONOUS_SUBMISSION = true';
```
Comment (williamjallen) [inline app/cdash/tests/test_enable_async.php:19]: I believe that you need to make this a double-quoted string to use the newline as expected here.  Escape sequences such as `\n` are only interpreted in double-quoted strings.

File: app/Console/Commands/MigrateConfig.php [hunk 4]
```diff
@@ -114,7 +118,11 @@ public function handle()
 
         // Read CDash config files, parsing their variables and changing
         // the names to Laravel equivalents as necessary.
-        include $cdash_app_dir . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'config.php';
+        $config_dir = $cdash_app_dir . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR;
```
Comment (zackgalbreath) [inline app/Console/Commands/MigrateConfig.php:121]:
```suggestion
```

(duplicate of line 83)

File: app/cdash/tests/test_autoremovebuilds_on_submit.php [hunk 5]
```diff
@@ -197,7 +158,7 @@ public function testBuildsRemovedOnSubmission()
         }
 
         // Make sure we didn't inadvertently delete the whole upload directory.
-        if (!file_exists("{$config->get('CDASH_ROOT_DIR')}/public/upload")) {
+        if (!file_exists(config('cdash.root_dir') . "/public/upload")) {
```
Comment (zackgalbreath) [inline app/cdash/tests/test_autoremovebuilds_on_submit.php:161]:
```suggestion
        if (!file_exists(base_path('app/cdash/public/upload')) {
```

File: config/cdash.php [hunk 6]
```diff
@@ -68,7 +71,10 @@
     'login_field' => env('LOGIN_FIELD', 'Email'),
     'max_upload_quota' => env('MAX_UPLOAD_QUOTA', 10),
     'notify_pull_request' => env('NOTIFY_PULL_REQUEST', false),
+    'php_exe' => env('PHP_EXE', ''),
+    'phpunit_exe' => env('PHPUNIT_EXE', ''),
     'remote_workers' => env('REMOTE_WORKERS', false),
+    'root_dir' => env('ROOT_DIR', $cdash),
```
Comment (zackgalbreath) [inline config/cdash.php:77]:
```suggestion
```

File: config/cdash.php [hunk 7]
```diff
@@ -52,12 +52,15 @@
         ],
     ],
     'active_project_days' => env('ACTIVE_PROJECT_DAYS', 7),
+    'asynchronous_submission' => env('CDASH_ASYNCHRONOUS_SUBMISSION', 'false'),
```
Comment (zackgalbreath) [inline config/cdash.php:55]:
```suggestion
```

(I don't think this setting is used any more...)

**Review by zackgalbreath — APPROVED** [review]

**Pull request merged**